api_v3/services/BulkUploadService.php

```diff
@@ -243,20 +243,39 @@ public function serveLogAction($id)
 	 */
 	public function abortAction($id)
 	{
-	    $c = new Criteria();
-	    $c->addAnd(BatchJobPeer::ID, $id);
+		$c = new Criteria();
+		$c->addAnd(BatchJobPeer::ID, $id);
 		$c->addAnd(BatchJobPeer::PARTNER_ID, $this->getPartnerId());
 		$c->addAnd(BatchJobPeer::JOB_TYPE, BatchJobType::BULKUPLOAD);
 		$batchJob = BatchJobPeer::doSelectOne($c);
-		
-	    if (!$batchJob)
-		    throw new KalturaAPIException(KalturaErrors::BULK_UPLOAD_NOT_FOUND, $id);
-		
-		kJobsManager::abortJob($id, BatchJobType::BULKUPLOAD, true);
-		
+
+		if (!$batchJob) {
+			$c = new Criteria();
+			$c->addAnd(BatchJobLogPeer::JOB_ID, $id);
+			$c->addAnd(BatchJobLogPeer::PARTNER_ID, $this->getPartnerId());
+			$c->addAnd(BatchJobLogPeer::JOB_TYPE, BatchJobType::BULKUPLOAD);
+
+			$crit = $c->getNewCriterion(BatchJobLogPeer::ABORT, null);
```

Comment:
To make the query more efficient, use $c->addAnd(BatchJobLogPeer::ABORT, BatchJobExecutionStatus::ABORTED, Criteria::NOT_EQUAL);